tests/Microsoft.Identity.Web.Test.Integration/AcquireTokenForAppIntegrationTests.cs

```diff
@@ -254,6 +254,47 @@ public async Task GetAccessTokenForApp_WithAnonymousController_Async()
             Assert.NotNull(token);
         }
 
+        [Fact]
+        public async Task GetAccessTokenForApp_ServiceProviderSetInExtraParameters()
+        {
+            // ASP.NET Core builder.
+            var serviceCollection = WebApplication.CreateBuilder().Services;
+            var configuration = new ConfigurationBuilder()
+                .AddInMemoryCollection(new Dictionary<string, string?>
+                {
+                    { "AzureAd:Instance", "https://login.microsoftonline.com/" },
+                    { "AzureAd:TenantId", TestConstants.ConfidentialClientLabTenant },
+                    { "AzureAd:ClientId", TestConstants.ConfidentialClientId },
+                    { "AzureAd:ClientSecret", _ccaSecret },
+                })
+                .Build();
+
+            serviceCollection.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                .AddMicrosoftIdentityWebApi(configuration)
+                .EnableTokenAcquisitionToCallDownstreamApi()
+                .AddInMemoryTokenCaches();
+
+            IServiceProvider? serviceProvider = null;
+
+            // Configure the extension option such that the event is subscribed to
+            // so the test can observe if the service provider is set in the extra parameters
+            serviceCollection.Configure<TokenAcquisitionExtensionOptions>(options =>
```

Comment:
Nit: You have called `AddAuthentication`, so I think you'd need to use the named options form:

```csharp
 serviceCollection.Configure<TokenAcquisitionExtensionOptions>(JwtBearerDefaults.AuthenticationScheme, options =>
```

Reply:
If there is no configured name, I believe it will use the default non-named options. I ran this and this did pass.

It is a little bit of an awkward place to test this, but it the integration tests are the best place I could find.